.github/workflows/cibuildwheel.yml

```diff
@@ -0,0 +1,180 @@
+name: Wheels
+
+on:
+  pull_request:
+  push:
+    tags:
+      - "v*"
+  release:
+    types:
+      - published
+
+permissions:
+  contents: read
+
+jobs:
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: 3.x
+
+      - name: Install APT packages
+        if: contains(${{ matrix.os }}, 'ubuntu')
+        run: |
+          sudo apt update
+          sudo apt install libhdf5-dev libnetcdf-dev
+
+      - name: Build sdist
+        run: >
+          pip install build
+          &&  python -m build --sdist . --outdir dist
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: pypi-artifacts
+          path: ${{ github.workspace }}/dist/*.tar.gz
+
+
+  build_bdist:
+    name: "Build ${{ matrix.os }} (${{ matrix.arch }}) wheels"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-22.04
+            arch: x86_64
+          # - os: ubuntu-22.04
+          #   arch: aarch64
+          - os: macos-14
+            arch: arm64
+            CIBW_ENVIRONMENT: MACOSX_DEPLOYMENT_TARGET=14.0
+          - os: macos-13
+            arch: x86_64
+            CIBW_ENVIRONMENT: MACOSX_DEPLOYMENT_TARGET=13.0
```

Comment:
These are a bit high, most folks set a lower bar with `MACOSX_DEPLOYMENT_TARGET=10.9`. If we aim to set 10.9 that we would need to build our own HDF5. I'm not  sure how many people we are leaving behind by setting a higher deployment target, probably the same amount as brew. With that said, I'd like to try the HDF5 in another PR. This one is getting unwieldy already.

Reply:
Looks like 13.0 was released in Oct 24, 2022 - this seems problematic to me.  I think we would be leaving at least half our users behind with this restriction.

Reply:
I have some guidelines on adding hdf5 from another project. I'll try that but I don't know if I can make it before the next release. Indeed, brew is too new.

Reply:
OK I guess we could release 1.7.0 with the 13.0 restriction, and add new wheels to the pypi release later one the hdf5 build is working?

Reply:
just a thought - [macports](https://ports.macports.org/port/hdf5/details/) supports older OSes than brew. Have no idea how hard that would be to get working.

Reply:
I'll try that tomorrow. I have two other options I want to experiment before trying to build it ourselves. One is to use an older mac image from GHA, the other is the pytables infrastructure to get hdf5 from source.

Note that I did try the conda-forge route, like we are doing for Windows, see https://github.com/ocefpaf/netcdf4-python/actions/runs/9471253609/job/26094347761

However, I don't think that is a good option here due to the big size of the wheel. Conda- forge packages are built with way too many bells and whistle and a wheel of ~24 M is not OK.

Reply:
agree that 24M is pretty bloated, but I'd rather have that than restrict to macos 13.0

Reply:
Let's keep that as a plan B then. I'll send if I can get a better alternative working first. Pytable's infrastructure seems pretty close to what we need here.

Reply:
Would MACOSX_DEPLOYMENT_TARGET 11 work for you? That is the oldest image available and brew is working fine to install both hdf5 and netcdf on it compiling from source. I still want to pursue the approach used in pytables at a later time. just want to be sure I'll have all the proper dependencies used in the current multibuild.

Reply:
a target of macos 11 would be fine, yes.

Reply:
> a target of macos 11 would be fine, yes.

Done!

> agree that 24M is pretty bloated, but I'd rather have that than restrict to macos 13.0

BTW, the main reason for that size was the inclusion of icu. We are discussing an alternative in conda-forge to remove that so we can make smaller wheels.

![image](https://github.com/Unidata/netcdf4-python/assets/950575/370ee8a1-48ce-481a-9109-6021999fa28e)

Reply:
thanks a million @ocefpaf!  Merging now...